### PR TITLE
Fix #10695 Handle empty values for and/or/not/nor filters

### LINKED ISSUE #10695: WFS requests from mapstore by default return a filter `<ogc:And>undefined</ogc:And>`
## How to reproduce

-  Open a map with WFS layer like this https://dev-mapstore.geosolutionsgroup.com/mapstore/#/viewer/10358
- Open the feature grid on the layer 
- See the WFS POST request body in developer tool

*Expected Result*
There should not a filter

*Current Result*
The filter contains `<ogc:Filter><ogc:And>undefined</ogc:And></ogc:Filter>` . Undefined string is tolerated by GeoServer, but not valid
 Image

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
This PR avoids the `<ogc:And>undefined</ogc:And>` in OGC filters that happens when having an empty filter.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#10695 

**What is the new behavior?**

Fix #10695 

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/utils/ogc/Filter/__tests__/operators-test.js [hunk 1]
```diff
@@ -72,6 +72,17 @@ describe('OGC Operators', () => {
...
+    it('logical functions with empty content', () => {
+        expect(logical.and("ogc"
+
+        )).toBe('');
+        expect(logical.or("ogc"
+
+        )).toBe('');
+        expect(logical.not("ogc"
+
+        )).toBe('');
```
Comment (MV88) [inline web/client/utils/ogc/Filter/__tests__/operators-test.js:84]: can't you have these in one line?

File: web/client/utils/ogc/Filter/__tests__/FilterBuilder-test.js [hunk 2]
```diff
@@ -79,6 +78,20 @@ describe('FilterBuilder', () => {
...
+        ).toBe("");
+        expect(
+            b.or()
+        ).toBe("");
+        expect(
+            b.not()
+        ).toBe("");
+        expect(
+            b.nor()
+        ).toBe("");
```
Comment (MV88) [inline web/client/utils/ogc/Filter/__tests__/FilterBuilder-test.js:94]: can't you have these in one line ?

File: web/client/utils/ogc/Filter/operators.js [hunk 3]
```diff
@@ -44,7 +44,7 @@ const upper = (ns, value) => `<${ns}:UpperBoundary>${value}</${ns}:UpperBoundary
  * @param {string|Array} content content
  * @returns the operation result
  */
-const multiop = (ns, op, content) => op(ns, Array.isArray(content) ? content.join("") : content);
+const multiop = (ns, op, content) => op(ns, Array.isArray(content) ? content.join("") : content );
```
Comment (MV88) [inline web/client/utils/ogc/Filter/operators.js:47]:
```suggestion
const multiop = (ns, op, content) => op(ns, Array.isArray(content) ? content.join("") : content);
```

**Review by MV88 — CHANGES_REQUESTED** [review]
LGTM

File: web/client/utils/ogc/Filter/__tests__/FilterBuilder-test.js (hunk 2 above)
Comment (offtherailz) [reply web/client/utils/ogc/Filter/__tests__/FilterBuilder-test.js:94]:
```suggestion
        expect(b.and()).toBe("");
        expect(b.or()).toBe("");
        expect(b.not()).toBe("");
        expect(b.nor()).toBe("");
```

File: web/client/utils/ogc/Filter/__tests__/operators-test.js (hunk 1 above)
Comment (offtherailz) [reply web/client/utils/ogc/Filter/__tests__/operators-test.js:84]:
```suggestion
        expect(logical.and("ogc")).toBe('');
        expect(logical.or("ogc")).toBe('');
        expect(logical.not("ogc")).toBe('');
```

**Review by MV88 — APPROVED** [review]

**Comment by tdipisa** [comment]
@ElenaGallo please check in DEV on deployed and let us know for the backport.

**Comment by ElenaGallo** [comment]
Test passed, @MV88 please backport. Thanks

**Pull request merged**